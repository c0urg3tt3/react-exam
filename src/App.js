import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { connect } from 'react-redux'

import './App.css'

import fetchJedi from './redux/actions/jedi/fetch'

import AppHeader from './components/app-header/component'
import AppLogo from './components/app-logo/component'
import AppTitle from './components/app-title/component'
import JediListItem from './components/jedi-list-item/component'
import JediListEmpty from './components/jedi-list-empty/component'

export class App extends Component {
  componentWillMount() {
    this.props.fetchJedi()
  }

  render() {
    const { jedies } = this.props

    return (
      <div className="app">
        <AppHeader>
          <AppLogo/>
          <AppTitle/>
        </AppHeader>
        {(!jedies.length && (
          <JediListEmpty>
          {({message, quote, autor}) => (
            <div className="jedi-quote">
              <p>{message}</p>
              <blockquote>
                <p>{quote}</p>
                <footer>{autor}</footer>
              </blockquote>
            </div>
          )}
          </JediListEmpty>
        ))
        || (
          <ul className="app-list">
          {
            jedies.map((jedi, index) => (
              <JediListItem key={index} jedi={jedi}/>
            ))
          }
          </ul>
        )}
      </div>
    )
  }
}

App.propTypes = {
  jedies: PropTypes.array,
  isFetching: PropTypes.bool,
  isFetchingError: PropTypes.bool,
  fetchJedi: PropTypes.func,
}

App.defaultProps = {
  jedies: [],
  isFetching: false,
  isFetchingError: true,
  fetchJedi: noop => noop
}

function mapStateToProps(state) {
  return { ...state.jedi.list }
}

function mapDispatchToProps(dispatch) {
  return {
    fetchJedi: () => dispatch(fetchJedi()),
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(App)
