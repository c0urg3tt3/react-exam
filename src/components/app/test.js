import React from 'react'
import { shallow } from 'enzyme'

import App from './component'

describe('[component] App', () => {
  it('should render component', () => {
    const wrapper = shallow(<App/>)

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('div')).toEqual(true)
    expect(wrapper.hasClass('app')).toEqual(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render children', () => {
    const wrapper = shallow(
      <App>
        <header>Sith List</header>
        <main>Who know ?</main>
      </App>
    )

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.find('header').text()).toEqual("Sith List")
    expect(wrapper.find('main').text()).toEqual("Who know ?")
    expect(wrapper.html()).toMatchSnapshot()
  })
})
