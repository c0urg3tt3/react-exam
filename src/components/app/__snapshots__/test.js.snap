// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] App should render children 1`] = `"<div class=\\"app\\"><header>Sith List</header><main>Who know ?</main></div>"`;

exports[`[component] App should render component 1`] = `"<div class=\\"app\\"></div>"`;
